crates/core/src/test.rs

```diff
@@ -15305,3 +15305,789 @@ fn or_file() {
+                |}
+                |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
+
+#[test]
+fn csharp_pattern_matching_is() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                |language csharp
+                |
+                |`if ($obj is $type)
+                |{
+                |    var $var = ($type)$obj;
+                |    $body
+                |}` => `if ($obj is $type $var)
+                |{
+                |    $body
+                |}`
+                |"#
+            .trim_margin()
+            .unwrap(),
+            source: r#"
+                |if (shape is Circle)
+                |{
+                |    var circle = (Circle)shape;
+                |    DrawCircle(circle.Radius);
+                |}
+                |"#
+            .trim_margin()
+            .unwrap(),
+            expected: r#"
+                |if (shape is Circle circle)
+                |{
+                |    DrawCircle(circle.Radius);
+                |}
+                |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
+
+#[test]
+fn csharp_linq_orderby() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                |language csharp
+                |
+                |`from $item in $collection
+                |orderby $item.$property select $item` =>
+                |                    `from $item in $collection
+                |                    orderby $item.$property descending
+                |                    select $item`
+                |"#
+            .trim_margin()
+            .unwrap(),
+            source: r#"
+                |public class Example {
+                |    public void Query() {
+                |        var result = from user in users
+                |                    orderby user.Name
+                |                    select user;
+                |    }
+                |}
+                |"#
+            .trim_margin()
+            .unwrap(),
+            expected: r#"
+                |public class Example {
+                |    public void Query() {
+                |        var result = from user in users
+                |                    orderby user.Name descending
+                |                    select user;
+                |    }
+                |}
+                |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
+
+#[test]
+fn csharp_groupby_keyword() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                |language csharp
+                |
+                |`from $item in $collection
+                |group $item by $item.$key` =>
+                |                    `from $item in $collection
+                |                    group $item by $item.$key into g
```

Comment:
We should also add tests for specific snippets *inside* a linq query.

Ex.

```
language csharp

`group $item by $item.key` => .
```

Reply:
fixed in 7ed9e1f30cfd65bc9fb4b5dd6f99b43b4c3300ee